test/Promise-test.js

```diff
@@ -14,32 +14,58 @@ describe('Promise', () => {
 	})
 
 	it('should reject if resolver throws synchronously', () => {
-		let expected = new Error()
+		const expected = new Error()
 		return new Promise(() => { throw expected })
 			.then(assert.ifError, x => assert.strictEqual(expected, x))
 	})
 
-	it('should fulfill with value', () => {
-		let expected = {}
-		return new Promise(resolve => resolve(expected))
-			.then(x => assert.strictEqual(expected, x))
-	})
+	describe('resolvers', () => {
+		it('should fulfill with value', () => {
+			const expected = {}
+			return new Promise(resolve => resolve(expected))
+				.then(x => assert.strictEqual(expected, x))
+		})
 
-	it('should resolve to fulfilled promise', () => {
-		let expected = {}
-		return new Promise(resolve => resolve(fulfill(expected)))
-			.then(x => assert.strictEqual(expected, x))
-	})
+		it('should resolve to fulfilled promise', () => {
+			const expected = {}
+			return new Promise(resolve => resolve(fulfill(expected)))
+				.then(x => assert.strictEqual(expected, x))
+		})
 
-	it('should resolve to rejected promise', () => {
-		let expected = {}
-		return new Promise(resolve => resolve(reject(expected)))
-			.then(assert.ifError, x => assert.strictEqual(expected, x))
-	})
+		it('should resolve to rejected promise', () => {
+			const expected = new Error()
+			return new Promise(resolve => resolve(reject(expected)))
+				.then(assert.ifError, x => assert.strictEqual(expected, x))
+		})
 
-	it('should reject with value', () => {
-		let expected = {}
-		return new Promise((resolve, reject) => reject(expected))
-			.then(assert.ifError, x => assert.strictEqual(expected, x))
+		it('should reject with value', () => {
+			const expected = new Error()
+			return new Promise((resolve, reject) => reject(expected))
+				.then(assert.ifError, x => assert.strictEqual(expected, x))
+		})
+
+		it('should asynchronously fulfill with value', () => {
```

Comment:
Yes, exactly. Calling methods on already resolved promises isn't very interesting, they always just call `near` and delegate the call - making the `future` tests test the same thing as the `fulfill`/`reject` ones.
Being resolved after the method invocation is the most common real-world usecase however.
I guess one could simplify the tests by checking the invariants on `_when` and the `Action` interface directly, but I didn't want to test internals.